wikibaseintegrator/entities/baseentity.py

```diff
@@ -299,6 +303,23 @@ def get_entity_url(self, wikibase_url: str | None = None) -> str:
 
         raise ValueError('wikibase_url or entity ID is null.')
 
+    def validate_schema(self, entity_schema: str, language: str | None = None) -> bool:
+        if isinstance(entity_schema, str):
+            pattern = re.compile(r'^(?:[a-zA-Z]+:)?E?([0-9]+)$')
+            matches = pattern.match(entity_schema)
+
+            if not matches:
+                raise ValueError(f"Invalid EntitySchema ID ({entity_schema}), format must be 'E[0-9]+'")
+
+            entity_schema = f'E{matches.group(1)}'
+        elif isinstance(entity_schema, int):
+            entity_schema = f'E{entity_schema}'
+        else:
+            raise ValueError(f"Invalid EntitySchema ID ({entity_schema}), format must be 'E[0-9]+'")
+
+        language = str(language or config['DEFAULT_LANGUAGE'])
+        return EntityShape(qid=self.id, eid=entity_schema, lang=language).get_result().is_valid
```

Comment:
I think the validation should return the result from EntityShape instead.
Then we provide an example to check if it was valid or not.